v2/internal/internalguantlettest/gauntlet_test.go

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package internalgauntlettest
+
+import (
+	"context"
+	"net"
+	"net/url"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/yarpc/v2"
+	"go.uber.org/yarpc/v2/yarpchttp"
+	"go.uber.org/yarpc/v2/yarpcrandpeer"
+	"go.uber.org/yarpc/v2/yarpcrouter"
+)
+
+type lifecycle interface {
+	Start(context.Context) error
+	Stop(context.Context) error
+}
+
+func newProcedures() []yarpc.TransportProcedure {
+	return jsonProcedures()
+}
+
+func newInbound(t *testing.T, transport string, listener net.Listener, procedures []yarpc.TransportProcedure) (stop func() error) {
+	var inbound lifecycle
+	router := yarpcrouter.NewMapRouter(_service, procedures)
+
+	switch transport {
+	case _http:
+		inbound = &yarpchttp.Inbound{
+			Listener: listener,
+			Router:   router,
+		}
+
```

Comment:
nit: We can remove the newline(s) between cases in the switch. Similar applies elsewhere.

Reply:
I actually prefer the separation to make larger blocks more distinguishable if that's ok

Reply:
Yea works for me :+1: